eng/tools/TypeSpecValidation/src/TypeSpecValidation.ts

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env node
+
+import {exec} from "child_process";
+import fs from "fs";
+import path from "path";
+import {simpleGit} from 'simple-git';
+
+async function runCmd(cmd:string, cwd:string) {
+    console.log(`run command:${cmd}`)
+    const { err, stdout, stderr } = await new Promise((res) =>
+        exec(
+        cmd,
+        { encoding: "utf8", maxBuffer: 1024 * 1024 * 64, cwd: cwd},
+        (err: unknown, stdout: unknown, stderr: unknown) =>
+            res({ err: err, stdout: stdout, stderr: stderr })
+        )
+    ) as any;
+    let resultString = stderr + stdout;
+    console.log("Stdout output:")
+    console.log(stdout)
+    if (stderr) {
+        console.log("Stderr output:")
+        console.log(stderr)
+    }
+    if (stderr || err) {
+        throw new Error(err);
+    }
+    return resultString as string;
+}
+    
+async function main() {
+    const args = process.argv.slice(2);
+    const folder = args[0];
```

Comment:
also maybe not needed now but node has a built-in utils do have a cli with flags 

https://nodejs.org/api/util.html#parseargs-tokens

using it from the start will make sure no hacky things get added later to add an flag

Reply:
I will change this, I agree might as well get it right now

Reply:
@ckairen: Please open an issue and we can fix this in a later PR

Reply:
Added in this PR